app/importers/unpaywall_publication_importer.rb

```diff
@@ -81,16 +91,27 @@ def update_publication(publication, unpaywall_json)
         locations_to_delete = existing_locations.reject { |l| unpaywall_locations_by_url.key? l.url }
         locations_to_delete.each(&:destroy)
 
-        publication.open_access_status = unpaywall_json['oa_status']
+        publication.open_access_status = if publication.doi.present?
+                                           unpaywall_json['oa_status']
+                                         else
+                                           unpaywall_json['results'].first['response']['oa_status']
```

Comment:
If we send a title search to unpaywall and it returns nothing, do we still receive the "results" key in the hash?  If it does return the "results" key, but there's nothing in it, we may run into some NoMethod for nil errors when trying to grab the 'response' and 'oa_status'.  If you run the imports locally, do you see a lot of these errors?

Reply:
I didn't run the imports locally because it seemed like it would take a few days. Since it matches on keywords, I don't think it would occur that often, but I can a nil check before assigning unpaywall_title just to be safe